GameProject/Assets/OpenWorld Assets/Scripts/trdprsnCharacter.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class trdprsnCharacter : MonoBehaviour
+{
+
+    [SerializeField] float movespeed;
+    [SerializeField] float rotatespeed;
+    //private float rotX;
+    //private float rotY;
+    [SerializeField] float rotZ;
+    private Camera cam;
+
+    private void Start()
+    {
+        cam = Camera.main;
+    }
+
+    void Update()
+    {
+        Playerturn(); // turn to face the player
+        Playerrotate(); // rotate the player itself
+    }
+
+    private void FixedUpdate()
+    {
+        Playermovement();
+    }
+
+    void Playermovement()
+    {
+        //forward and back controls
+        if (Input.GetKey(KeyCode.LeftShift) && Input.GetKey("w")) // if w and shift are pressed
+        {
+            transform.position += transform.TransformDirection(Vector3.forward) * Time.deltaTime * movespeed * 2.5f;
+        }
+        else if (Input.GetKey("w") && !Input.GetKey(KeyCode.LeftShift)) // if only w is pressed
+        {
+            transform.position += transform.TransformDirection(Vector3.forward) * Time.deltaTime * movespeed;
+        }
+        else if (Input.GetKey("s") && !Input.GetKey(KeyCode.LeftShift)) // if only s is pressed
+        {
+            transform.position -= transform.TransformDirection(Vector3.forward) * Time.deltaTime * movespeed;
+        }
+
+        //left and right controls
+        if (Input.GetKey("a") && !Input.GetKey("d")) // if only a is pressed
+        {
+            transform.position += transform.TransformDirection(Vector3.left) * Time.deltaTime * movespeed;
+        }
+        else if (Input.GetKey("d") && !Input.GetKey("a")) // if only d is pressed
```

Comment:
All of the above `GetKey` calls rely on strings and not the `KeyCode` enum. Please rectify this.